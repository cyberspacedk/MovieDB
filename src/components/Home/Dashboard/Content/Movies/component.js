/* eslint-disable react/jsx-key */
/* eslint-disable react/prop-types */
import React from 'react';
import { Layout, Row, Col, Pagination, BackTop } from 'antd';
import Card from '../../../../shared/Card';

// eslint-disable-next-line no-unused-vars
const Movies = ({ response, totalResults, history }) => (
  <Layout>
    <Layout.Content>
      <div className="top-margin">
        <Row type="flex" gutter={16} justify="center">
          <Col span={20}>
            {response &&
              response.map(item => (
                <Card
                  key={item.id}
                  id={item.id}
                  poster={item.poster_path}
                  title={item.original_title}
                  overview={item.overview}
                  history={history}
                />
              ))}
          </Col>
        </Row>

        {totalResults && (
          <Row type="flex" justify="center">
            <Col>
              <Pagination
                defaultCurrent={1}
                total={totalResults}
                className="pagination"
                defaultPageSize={20}
              />
            </Col>
          </Row>
        )}
      </div>
    </Layout.Content>
    <BackTop />
  </Layout>
);

export default Movies;
