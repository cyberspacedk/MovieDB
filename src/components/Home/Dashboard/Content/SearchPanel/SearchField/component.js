import React from 'react';
import PropTypes from 'prop-types';
import { Input, Form } from 'antd';

const { Item } = Form;
const { Search } = Input;

const SearchField = ({
  field,
  form: { touched, errors, handleSubmit },
  ...props
}) => (
  <Item
    validateStatus={touched[field.name] && errors[field.name] && 'error'}
    help={errors[field.name]}
  >
    <Search
      onSubmit={handleSubmit}
      placeholder="Enter movie name"
      size="large"
      enterButton="Search"
      className="top-margin"
      onSearch={handleSubmit}
      {...field}
      {...props}
    />
  </Item>
);

SearchField.propTypes = {
  field: PropTypes.shape({ name: PropTypes.string }),
  form: PropTypes.shape({
    touched: PropTypes.object,
    errors: PropTypes.object,
  }),
};

export default SearchField;
