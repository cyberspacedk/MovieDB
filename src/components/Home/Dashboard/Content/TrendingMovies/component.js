/* eslint-disable react/prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import { Layout, Row, Col, Card, BackTop } from 'antd';
import Spinner from '../../../../shared/StatusFields/Spinner';

const TrendingMovies = ({ loading, topFilms, singleFilmRequest, history }) => (
  <Layout>
    <Layout.Content>
      <div className="top-margin">
        <Row type="flex" gutter={16} justify="center" align="left">
          <Col span={20}>
            {loading ? (
              <Spinner />
            ) : (
              topFilms.map(item => (
                <Col
                  key={item.id}
                  xs={{ span: 12 }}
                  sm={{ span: 12 }}
                  md={{ span: 8 }}
                  lg={{ span: 6 }}
                  xl={{ span: 4 }}
                  onClick={() => {
                    singleFilmRequest(item.id);
                    history.push(`/${item.id}`);
                  }}
                >
                  <Card
                    hoverable
                    cover={
                      <img
                        alt={item.title}
                        src={`https://image.tmdb.org/t/p/w200${
                          item.poster_path
                        }`}
                      />
                    }
                    className="top-margin card-film"
                    /*  actions={} */
                  >
                    <Card.Meta
                      title={item.original_title}
                      description={`${item.overview.slice(0, 100)}...`}
                    />
                  </Card>
                </Col>
              ))
            )}
          </Col>
        </Row>
      </div>
    </Layout.Content>
    <BackTop />
  </Layout>
);

TrendingMovies.propTypes = {
  loading: PropTypes.bool.isRequired,
  topFilms: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default TrendingMovies;
