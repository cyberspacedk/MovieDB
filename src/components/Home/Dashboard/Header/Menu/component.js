import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Typography, Col, Avatar, Dropdown, Icon, Menu } from 'antd';

const { Item, Divider } = Menu;
const { Text } = Typography;

const UserMenu = ({ username, authLogout, history }) => {
  const Overlay = () => (
    <Menu>
      <Item>
        <Link to="/">Dashboard</Link>
      </Item>
      <Divider />
      <Item>
        <Link to="/lists">My Lists</Link>
      </Item>
      <Item>
        <Link to="/watchlist">Watchlist</Link>
      </Item>
      <Item>
        <Link to="/favorites">Favorites</Link>
      </Item>
      <Divider />
      <Item
        onClick={() => {
          authLogout();
          history.push('/');
        }}
      >
        Logout
      </Item>
    </Menu>
  );

  return (
    <Col>
      <Avatar icon="user" />
      <Dropdown overlay={Overlay}>
        <Text>
          {username}
          <Icon type="caret-down" />
        </Text>
      </Dropdown>
    </Col>
  );
};

UserMenu.propTypes = {
  username: PropTypes.string.isRequired,
  authLogout: PropTypes.func.isRequired,
  history: PropTypes.shape(PropTypes.object).isRequired,
};

export default UserMenu;
