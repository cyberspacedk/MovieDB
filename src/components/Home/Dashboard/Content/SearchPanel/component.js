import React from 'react';
import { Form, Field } from 'formik';
import { Layout, Row, Col } from 'antd';
import SearchField from './SearchField/component';

const SearchPanel = () => (
  <Layout.Content>
    <Row type="flex">
      <Col
        xs={{ span: 22, offset: 1 }}
        sm={{ span: 20, offset: 2 }}
        md={{ span: 18, offset: 3 }}
        lg={{ span: 16, offset: 4 }}
        xl={{ span: 14, offset: 5 }}
      >
        <Form>
          <Field name="search" component={SearchField} />
        </Form>
      </Col>
    </Row>
  </Layout.Content>
);

export default SearchPanel;
