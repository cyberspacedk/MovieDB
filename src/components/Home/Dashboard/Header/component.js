/* eslint-disable react/prop-types */
import React from 'react';
import { Typography, Row, Col, Layout } from 'antd';
import Menu from './Menu';

const Header = () => {
  return (
    <Layout.Header>
      <Row type="flex" justify="space-between">
        <Col>
          <Typography.Text>THE MOVIE DB</Typography.Text>
        </Col>
        <Menu />
      </Row>
    </Layout.Header>
  );
};

export default Header;
