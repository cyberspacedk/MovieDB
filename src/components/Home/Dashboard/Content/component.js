import React from 'react';
import PropTypes from 'prop-types';
import SearchPanel from './SearchPanel';
import TrendingMovies from './TrendingMovies';
import Movies from './Movies';

export const PageContent = ({ searchResponse }) => {
  return (
    <>
      <SearchPanel />
      {searchResponse ? <Movies /> : <TrendingMovies />}
    </>
  );
};

PageContent.propTypes = {
  searchResponse: PropTypes.number.isRequired,
};

export default PageContent;
