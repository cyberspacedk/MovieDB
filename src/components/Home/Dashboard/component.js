import React from 'react';
import { Switch, Route } from 'react-router-dom';
import Header from './Header';
import Content from './Content';
import Movie from '../../Movie';

const Dashboard = () => {
  return (
    <div>
      <Header />
      <Switch>
        <Route exact path="/" component={Content} />
        <Route path="/:id" component={Movie} />
      </Switch>
    </div>
  );
};

export default Dashboard;
