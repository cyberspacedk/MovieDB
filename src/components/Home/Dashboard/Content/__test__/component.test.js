import React from 'react';
import { shallow } from 'enzyme';
import { PageContent } from '../component';

describe('Component: PageContent', () => {
  const props = {
    searchResponse: 9,
  };
  const container = shallow(<PageContent {...props} />);

  it('Snapshot: should match. Display search results', () => {
    expect(container).toMatchSnapshot();
  });

  it('Snapshot: should match. Display trending', () => {
    const nextProps = {
      searchResponse: 0,
    };
    const container = shallow(<PageContent {...nextProps} />);

    expect(container).toMatchSnapshot();
  });
});
