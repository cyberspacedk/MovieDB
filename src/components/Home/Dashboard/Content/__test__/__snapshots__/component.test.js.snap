// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: PageContent Snapshot: should match. Display search results 1`] = `
<Fragment>
  <Connect(WithFormik(SearchPanel)) />
  <Connect(withRouter(MoviesContainer)) />
</Fragment>
`;

exports[`Component: PageContent Snapshot: should match. Display trending 1`] = `
<Fragment>
  <Connect(WithFormik(SearchPanel)) />
  <Connect(withRouter(TrendingMoviesContainer)) />
</Fragment>
`;
